### PR TITLE
Fix clippy_extra.yml CI on billeo-engine side

### PR DESCRIPTION
**Pull request by n-eq**
* Check host OS is MacOS before calling NDK-related stuff (see #38)
* Use ONLY_CARGO_CHECK env variable to skip copying libs that don't exist

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -38,7 +38,15 @@ static CBL_LIB_DIR: &str = "libcblite-3.0.3/lib";
...
     generate_bindings()?;
     configure_rustc()?;
-    copy_lib()?;
+
+    // if we're currently in a cargo check workflow, no need to try to copy libs around.
+    // In particular, it won't work using the current clippy_extra on billeo-engine side
+    // because the used target is x86_64-linux-android, for which there are no libs under
+    // libcblite-3.0.3/lib folder.
+    // Note: ONLY_CARGO_CHECK is defined as "true" in clippy_extra.yml.
+    if env::var("ONLY_CARGO_CHECK").unwrap_or_default() != String::from("true") {
```
Comment (n-eq) [inline build.rs:47]:
```suggestion
    if env::var("ONLY_CARGO_CHECK").unwrap_or_default() != *"true" {
```

File: build.rs [hunk 2]
```diff
@@ -38,7 +38,15 @@ static CBL_LIB_DIR: &str = "libcblite-3.0.3/lib";
 fn main() -> Result<(), Box<dyn Error>> {
     generate_bindings()?;
     configure_rustc()?;
-    copy_lib()?;
+
+    // if we're currently in a cargo check workflow, no need to try to copy libs around.
+    // In particular, it won't work using the current clippy_extra on billeo-engine side
```
Comment (Nao-ris) [inline build.rs:43]: Please no information about billeo-engine, the goal is for this repo to be open source :-)

Comment (n-eq) [reply build.rs:43]: Ok, should I just remove the comment then?

Comment (Nao-ris) [reply build.rs:43]: You can, or write a simpler one that just points out that copying libs is not necessary in the context of a cargo check.
As you prefer :-)

**Review by Nao-ris — APPROVED** [review]

**Pull request merged**